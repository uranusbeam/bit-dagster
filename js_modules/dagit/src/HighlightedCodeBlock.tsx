import * as React from "react";
import { configure, highlightBlock } from "highlight.js";

import "highlight.js/styles/xcode.css";

export class HighlightedCodeBlock extends React.Component<{
  value: string;
  languages: string[];
  style?: React.CSSProperties;
}> {
  _el = React.createRef<HTMLPreElement>();

  componentDidMount() {
    if (this._el.current) {
      configure({ languages: this.props.languages });
      highlightBlock(this._el.current);
    }
  }

  render() {
    const { value, ...rest } = this.props;
    return (
      <pre ref={this._el} {...rest}>
        {value}
      </pre>
    );
  }
}
