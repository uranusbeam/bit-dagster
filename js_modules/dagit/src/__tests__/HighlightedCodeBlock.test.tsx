import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { HighlightedCodeBlock } from "../HighlightedCodeBlock";

it("correctly renders keywords inside a comment", () => {
  const component = TestRenderer.create(
    <BrowserRouter>
      <HighlightedCodeBlock
        languages={["sql"]}
        value={
          "-- this comment contains a dash - do not highlight if drop table\n" +
          "select * from q2_on_time_data\n" +
          "where origin = 'SFO'"
        }
        style={{
          height: 510,
          margin: 0,
          overflow: "scroll",
          fontSize: "0.9em"
        }}
      />
    </BrowserRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
