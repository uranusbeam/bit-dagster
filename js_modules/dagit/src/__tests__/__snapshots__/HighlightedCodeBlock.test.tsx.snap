// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders keywords inside a comment 1`] = `
<pre
  languages={
    Array [
      "sql",
    ]
  }
  style={
    Object {
      "fontSize": "0.9em",
      "height": 510,
      "margin": 0,
      "overflow": "scroll",
    }
  }
>
  -- this comment contains a dash - do not highlight if drop table
select * from q2_on_time_data
where origin = 'SFO'
</pre>
`;
